### PR TITLE
fix: hide unsuccessful profile image

### PR DESCRIPTION
**Pull request by adong**
## Context

We will hide the Github avatar url if the image loads unsuccessfully. 

~One suspect is that Chrome had removed `Referrers` header and causes GHE to render something like this, and now Firefox follows suit, and is removing `Referrers` header as well.~

Before:
image

## Objective

This `onerror="this.style.visibility='hidden';"` not working, because GHE returns http status with `200` for 0 bytes

image

Now it hides the avatar for [GHE](https://github.com/enterprise).

After:
image

## References

~[Firefox 87 trims HTTP Referrers by default to protect user privacy - Mozilla Security Blog](https://blog.mozilla.org/security/2021/03/22/firefox-87-trims-http-referrers-by-default-to-protect-user-privacy/)~
See details in: 
- [Avatar images are broken on GHE Server running in private mode](https://github.com/Reviewable/Reviewable/issues/770)
- [Cannot load avatars from GitHub Enterprise URLs](https://github.com/microsoft/vscode-pull-request-github/issues/416)
- [Preventing load of avatars on GitHub Enterprise by StanleyGoldman ](https://github.com/microsoft/vscode-pull-request-github/pull/1148) summarized accurately:

> After some research about GitHub Enterprise, I've learned that it will never return an avatar over the avatar url. The authentication needed for this is a cookie created by the GitHub Enterprise instance, we only have an OAuth Token.

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by jithine — APPROVED** [review]

**Review by tkyi — APPROVED** [review]

File: app/components/user-link/component.js [hunk 1]
```diff
@@ -1,5 +1,11 @@
 import Component from '@ember/component';
+import { computed } from '@ember/object';
+import ENV from 'screwdriver-ui/config/environment';
 
 export default Component.extend({
-  classNameBindings: ['large']
+  classNameBindings: ['large'],
+
+  isOpenSourceInstance: computed(function() {
+    return ENV.SDAPI_HOSTNAME === 'https://api.screwdriver.cd';
```
Comment (tkyi) [inline app/components/user-link/component.js:9]: maybe use a constant?

Comment (jithine) [reply app/components/user-link/component.js:9]: we should just have a flag, say ENV.SHOW_AVATARS : true/false Let's not hardcode host names

Comment (adong) [reply app/components/user-link/component.js:9]: Both GHE and Github versions are `production` environment. Current implementation auto detects.

If we make it like a feature flag, would this flag manually set by Cluster Admins?

Comment (jithine) [reply app/components/user-link/component.js:9]: yes, lets cluster admins deal with this. 

https://github.com/screwdriver-cd/ui/blob/57c043d4f3c923d89ec4de0656969563e5066cca/app/instance-initializers/supplementary-config.js#L7
https://github.com/screwdriver-cd/ui/blob/57c043d4f3c923d89ec4de0656969563e5066cca/nginx.conf#L93

Comment (adong) [reply app/components/user-link/component.js]: OK. In this case, we also need to update the Guide: https://github.com/screwdriver-cd/guide
to reflect on https://docs.screwdriver.cd/

File: app/instance-initializers/supplementary-config.js [hunk 2]
```diff
@@ -29,6 +29,9 @@ export function initialize() {
   if (window.SUPPLEMENTARY_CONFIG && window.SUPPLEMENTARY_CONFIG.RELEASE_VERSION) {
     ENV.APP.RELEASE_VERSION = window.SUPPLEMENTARY_CONFIG.RELEASE_VERSION;
   }
+  if (window.SUPPLEMENTARY_CONFIG && window.SUPPLEMENTARY_CONFIG.SHOW_AVATAR) {
+    ENV.APP.SHOW_AVATAR = window.SUPPLEMENTARY_CONFIG.SHOW_AVATAR;
```
Comment (jithine) [inline app/instance-initializers/supplementary-config.js:33]: need to add to https://github.com/screwdriver-cd/ui/blob/master/nginx.conf#L93 for this to work.

**Review by jithine — APPROVED** [review]

**Pull request merged**